soc/arm/st_stm32/stm32h7/soc.h

```diff
@@ -54,7 +54,7 @@
 #include <stm32h7xx_ll_usart.h>
 #endif /* CONFIG_SERIAL_HAS_DRIVER */
 
-#ifdef CONFIG_HWINFO_STM32
+#if defined(CONFIG_HWINFO_STM32) || defined(CONFIG_CLOCK_CONTROL)
```

Comment:
```suggestion
#if defined(CONFIG_HWINFO_STM32) || defined(CONFIG_CLOCK_CONTROL_STM32_CUBE)
```

Reply:
It is safer to use STM32 dedicated flags (less likely to be enabled accidentally)

Reply:
Done, you are right, that's safer !